Adding TS code foundations (#1581)
temporalio/documentation: docs-src/typescript/how-to-customize-activity-type-in-typescript.md

---
id: how-to-customize-activity-type-in-typescript
title: How to customize Activity Type in TypeScript
sidebar_label: Customize Activity Type
description: Customize Activity Type
tags:
  - developer-guide
  - typescript
---

> In TypeScript, to customize an Activity Name set the function name to your desired name.
> 
> In the following example the Activity Name is `greet`.

Reviewer: ```suggestion
In TypeScript, the Activity Type is the function name.

In the following example, the Activity Type is `greet`.
```

you can't customize this in TS

> In TypeScript, to customize an Activity Name set the function name to your desired name.
> 
> In the following example the Activity Name is `greet`.
> 
> <!--SNIPSTART typescript-activity-fn -->
> <!--SNIPEND-->

Reviewer: ```suggestion
In TypeScript, the Activity Type is the property of the [`WorkerOptions.activities`](https://typescript.temporal.io/api/interfaces/worker.workeroptions/#activities) object. 
Usually, that's the Activity function's name. 
For example, in the following code, the Activity Type is `greet`:

<!--SNIPSTART typescript-activity-fn -->
<!--SNIPEND-->

<!--SNIPSTART typescript-worker-create -->
<!--SNIPEND-->

You can customize the Activity Type by choosing a property other than the function name. In the following example, the Activity Type is `activityFoo`:

<!--SNIPSTART typescript-custom-activity-type -->
<!--SNIPEND-->
```
